packages/cli/src/Server.ts

```diff
@@ -779,236 +798,259 @@ class App {
 		// Returns workflows
 		this.app.get(
 			`/${this.restEndpoint}/workflows`,
-			ResponseHelper.send(async (req: express.Request, res: express.Response) => {
-				const queryBuilder = Db.collections.Workflow!.createQueryBuilder('w');
-				queryBuilder.select(['w.id', 'w.name', 'w.active', 'w.createdAt', 'w.updatedAt']);
-				queryBuilder.leftJoinAndSelect('w.tags', 't');
-
-				if (req.query.filter) {
-					const jsonFilters = JSON.parse(req.query.filter as string);
-					const keys = Object.keys(jsonFilters);
-					keys.forEach((key) => {
-						queryBuilder.andWhere(`w.${key} = :${key}`, { [key]: jsonFilters[key] });
+			ResponseHelper.send(async (req: WorkflowRequest.GetAll) => {
+				let workflows: WorkflowEntity[] = [];
+
+				if (req.user.globalRole.name === 'owner') {
+					workflows = await Db.collections.Workflow!.find({
+						select: ['id', 'name', 'active', 'createdAt', 'updatedAt'],
+						relations: ['tags'],
+						where: req.query,
+					});
+				} else {
+					const shared = await Db.collections.SharedWorkflow!.find({
+						relations: ['workflow', 'workflow.tags'],
+						where: buildWhereClause({
+							user: req.user,
+							entityType: 'workflow',
+						}),
+					});
+
+					if (!shared.length) {
+						throw new ResponseHelper.ResponseError('No workflows found', undefined, 404);
+					}
+
+					workflows = await Db.collections.Workflow!.find({
+						relations: ['tags'],
+						select: ['id', 'name', 'active', 'createdAt', 'updatedAt'],
+						where: {
+							id: In(shared.map(({ workflow }) => workflow.id)),
+							...req.query,
+						},
```

Comment:
Without using the query builder, as far as I found it is not possible to select a column from a relation when using `SharedWorkflow` as repository, or to fetch multiple sharings when using `Workflow` as repository. Hence the two calls to `find()`.